include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -171,6 +171,13 @@ def NonLocalAnchor : FIRRTLOp<"nla",
     /// Return true if any update is made.
     bool updateModule(StringAttr oldMod, StringAttr newMod);
 
+    /// Replace the oldMod module with newMod module in the namepath of the NLA.
+    /// Given the renameMap of old sym names to new sym names, update the
```

Comment:
Can you clarify?  What is the rename map and what is it used for?  And why is it dependent on leafiness of names?

Reply:
If a module is updated in the namepath, then the InnerRef to the symbol inside the old module also needs to be updated to the corresponding symbol in the new module. The rename map which is used by the dedup pass, is the map of the old symbols to the new symbols in the deduped module.